components/builder-events/src/connection.rs

```diff
@@ -0,0 +1,59 @@
+use crate::kafka::KafkaProducer;
+use std::{convert::From,
+          time::Duration};
+
+#[derive(Clone, Debug, Deserialize)]
+#[non_exhaustive]
+pub enum Provider {
+    Kafka,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(default)]
+pub struct EventBusCfg {
+    pub api_key:                String,
+    pub api_secret_key:         String,
+    pub bootstrap_nodes:        Vec<String>,
+    pub client_id:              String,
+    #[serde(with = "deserialize_into_duration")]
+    pub connection_retry_delay: Duration,
+    pub message_timeout_ms:     u64,
+    pub provider:               Provider,
+}
+
+impl Default for EventBusCfg {
+    fn default() -> Self {
+        EventBusCfg { api_key:                String::from("CHANGEME"),
+                      api_secret_key:         String::from("CHANGEMETOO"),
+                      bootstrap_nodes:        vec![String::from("localhost:9092")],
+                      client_id:              String::from("http://localhost"),
+                      connection_retry_delay: Duration::from_secs(3),
+                      message_timeout_ms:     3000,
+                      provider:               Provider::Kafka, }
+    }
```

Comment:
I was thinking that this could be a blanket struct for handling _any_ event bus. Granted, it's Kafka oriented right now, but could be extended to support other message platforms. Should that scenario arise, the operator will merely set `provider = <the_new_provider>` in the `config.toml` and set whichever new fields are necessary  for that bus type.

Reply:
I have broken out the configuration into Bus-specific fields in the `EventBusConfig` struct in https://github.com/habitat-sh/builder/pull/1540/commits/9f5f1e41cf492cd955da60b6f382128239b3a073

This way we have specific name-spaced configuration that is 100% in alignment with the bus-type selected for use.

```
impl Default for Provider {
    fn default() -> Provider { Provider::None }
}

/// Holds the configuration defining the EventBus provider bindings to load as well
/// as the specific configuration required for each message bus binding type.
#[derive(Clone, Debug, Deserialize)]
#[serde(default)]
pub struct EventBusConfig {
    #[serde(with = "util::serde::string")]
    pub provider: Provider,
    pub kafka:    KafkaConfig,
}

impl Default for EventBusConfig {
    fn default() -> Self {
        EventBusConfig { provider: Provider::default(),
                         kafka:    KafkaConfig::default(), }
    }
}
```

We still have to have defaults for API key and API secret key to satisfy the `Deserialize` trait. However, it should be pretty obvious in the logs that either the API key or secret are incorrect and need to be configured for the user's message bus and we certainly can't know the correct values in advance.